bin/cron/pardot_sync

```diff
@@ -0,0 +1,419 @@
+#!/usr/bin/env ruby
+require File.expand_path('../../../pegasus/src/env', __FILE__)
+require_relative '../../lib/state_abbr'
+require src_dir 'database'
+require_relative('../../dashboard/config/environment')
+require 'cdo/properties'
+require 'json'
+
+DASHBOARD_REPORTING_DB_READONLY = sequel_connect(
+  CDO.dashboard_reporting_db_reader, CDO.dashboard_reporting_db_reader)
+
+UPDATE_BATCH_SIZE = 100
+
+DASHBOARD_DB_NAME = "dashboard_#{Rails.env}"
+
+def log(s)
+  puts s      # emit to stdout
+  CDO.log.info s   # emit to log file
+end
+
+def truncate_contacts_rollup
+  start = Time.now
+  log "Truncating contacts_rollup"
+  PEGASUS_DB.run "TRUNCATE TABLE contacts_rollup"
+  log "   Completed in " + ((Time.now) - start).to_s + " seconds"
+end
+
+def insert_from_dashboard_contacts
+  start = Time.now
+  log "Inserting teacher contacts and IP geo data from dashboard.users"
+  PEGASUS_DB.run "
+    INSERT INTO contacts_rollup (email,name,dashboard_user_id,roles, city, state, postal_code, country)
+	  SELECT email COLLATE utf8_general_ci, name, users.id, 'Teacher', city, state, postal_code, country FROM #{DASHBOARD_DB_NAME}.users AS users
+    LEFT OUTER JOIN #{DASHBOARD_DB_NAME}.user_geos AS user_geos ON user_geos.user_id = users.id
+    WHERE users.user_type = 'teacher'AND LENGTH(email) > 0
+    ON DUPLICATE KEY UPDATE contacts_rollup.email=VALUES(email), contacts_rollup.name=VALUES(name), contacts_rollup.dashboard_user_id=VALUES(dashboard_user_id), roles=VALUES(roles)"
+  log "   Completed in " + ((Time.now) - start).to_s + " seconds"
+end
+
+def insert_from_dashboard_pd_enrollments
+  start = Time.now
+  log "Inserting contacts from dashboard.pd_enrollments"
+  PEGASUS_DB.run "
+    INSERT INTO contacts_rollup (email,name)
+	  SELECT email,name FROM dashboard_development.pd_enrollments AS pd_enrollments
+    WHERE LENGTH(pd_enrollments.email) > 0
+	  ON DUPLICATE KEY UPDATE email=contacts_rollup.email,name=contacts_rollup.name"
+  log "   Completed in " + ((Time.now) - start).to_s + " seconds"
+end
+
+def insert_from_pegasus_contacts
+  start = Time.now
+  log "Inserting contacts from pegasus.contacts"
+  PEGASUS_DB.run "
+    INSERT INTO contacts_rollup (email, name)
+    SELECT email,name FROM contacts WHERE LENGTH(email) > 0
+    ON DUPLICATE KEY UPDATE contacts_rollup.email=VALUES(email), contacts_rollup.name=VALUES(name)"
+  log "   Completed in " + ((Time.now) - start).to_s + " seconds"
+end
+
+def insert_from_pegasus_forms
+  start = Time.now
+  log "Inserting contacts and IP geo data from pegasus.forms"
+  PEGASUS_DB.run "
+    INSERT IGNORE INTO contacts_rollup (email, name, city, state, postal_code, country)
+    SELECT email,name, city, state, postal_code,country FROM forms
+    LEFT OUTER JOIN form_geos on form_geos.form_id = forms.id"
```

Comment:
Do you want to restrict (via `form_geos.indexed_at`) to those `form_geos` that have been geolocated?